### PR TITLE
fixed #11744

### LINKED ISSUE #11744: ExternalMessages - The "Date Sample was collected" is not changing if it is already pre-filled with a value from lab message
### Bug Description

Changing in UI the Sample data was collected is still taking the previous value that comes from the lab message

### Reproduction Steps

1. Login as NatUser + Admin
2. Have a lab message that contains a Sample data was collected, and Date and time of result to be before Sample data was collected
3. Fetch the message
4. Start processing the message
5. At the step of Sample don't change any dates and try to save it
6. A pop-up error message appears. related to the Date and time of result that should not be before Sample data was collected
7. Change the date in the Sample for Sample data was collected to be a correct one in relation with Date and time of result
8. Try to save again the Sample

### Expected Behavior

The sample should be saved successfully and the lab message to appear as processed

### Affected Area(s)

Web app

### SORMAS Version

1.82.0-SNAPSHOT (4f77f37)

### Android version/Browser

-

### User Role(s)

NatUser + Admin

### Server URL

test-de2

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ChristopherRiedel**
Fixes #11744

### CODE REVIEW

File: sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestForm.java [hunk 1]
```diff
@@ -307,12 +320,17 @@ protected void addFields() {
 			}
 		});
 
-		if (sample.getSamplePurpose() != SamplePurpose.INTERNAL) { // this only works for already saved samples
+		if (SamplePurpose.INTERNAL
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestForm.java:323]: please create a function similar to `getSampleDate` to simplify this condition

Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestForm.java:323]: please create a function similar to `getSampleDate` to simplify this condition

**Review by leventegal-she — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11768

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11768

**Pull request merged**